dn.go

```diff
@@ -143,6 +143,9 @@ func ParseDN(str string) (*DN, error) {
 			}
 		} else if char == ',' || char == '+' {
 			// We're done with this RDN or value, push it
+			if len(attribute.Type) == 0 {
+				return nil, errors.New("incomplete type, value pair")
```

Comment:
just had a thought... should we include `stringFromBuffer()` in the error to aid debugging the bad attribute? or if we don't want to allow injecting arbitrary values into the message, at least include the position?